src/core/thread/address_resolver.cpp

```diff
@@ -518,19 +575,22 @@ void AddressResolver::HandleAddressQuery(Coap::Header &aHeader, Message &aMessag
     ThreadLastTransactionTimeTlv lastTransactionTimeTlv;
     Child *children;
     uint8_t numChildren;
+    char stringBuffer[Ip6::Address::kIp6AddressStringSize];
 
     VerifyOrExit(aHeader.GetType() == OT_COAP_TYPE_NON_CONFIRMABLE &&
                  aHeader.GetCode() == OT_COAP_CODE_POST);
 
-    otLogInfoArp(GetInstance(), "Received address query from %04x", HostSwap16(aMessageInfo.GetPeerAddr().mFields.m16[7]));
-
     SuccessOrExit(ThreadTlv::GetTlv(aMessage, ThreadTlv::kTarget, sizeof(targetTlv), targetTlv));
     VerifyOrExit(targetTlv.IsValid());
 
     mlIidTlv.Init();
 
     lastTransactionTimeTlv.Init();
 
+    otLogInfoArp(GetInstance(), "Received address query from %04x for target %s",
```

Comment:
nit: be consistent on the `0x` prefix.

Reply:
good catch. fixed it in new pushed commit.